src/types/mod.rs

```diff
@@ -79,9 +79,9 @@ macro_rules! pyobject_native_type_named (
 
 #[macro_export]
 macro_rules! pyobject_native_type (
```

Comment:
Since most native types has `builtin` as their module name, I suggest to make the argument `$module` optional.
E.g., you can do it by 
```rust
#[macro_export]
macro_rules! pyobject_native_type (
    ($name: ty, $typeobject: expr, $module: expr, $checkfunction: path $(,$type_param: ident)*) => {
        ...
    };
    ($name: ty, $typeobject: expr, $checkfunction: path $(,$type_param: ident)*) => {
        pyobject_native_type!{$name, $typeobject, Some("builtin"), $checkfunction $(,$type_param)*}
    };
);
```

Reply:
I tried to do that at first (mostly to save me from editing all `pyobject_native_type` invocations) but it doesn't work since the resulting macro is ambiguous: since a path is also a valid expression, 
```rust
pyobject_native_type!(a, b, c, d)
```
could be
```rust
pyobject_native_type!($name = a, $typeobject = b, $module = c, $checkfunction = d)
```
or 
```rust
pyobject_native_type!($name = a, $typeobject = b, $checkfunction = c, $type_param = d)
```

Reply:
I tried both
```rust


#[macro_export]
macro_rules! pyobject_native_type (
    ($name: ty, $typeobject: expr, $checkfunction: path $(,$type_param: ident)*) => {
        pyobject_native_type!{$name, $typeobject, Some("builtin"), $checkfunction $(,$type_param)*}
    };
    ($name: ty, $typeobject: expr, $module: expr, $checkfunction: path $(,$type_param: ident)*) => {
        ...
    };
);
```
and
```rust
    ($name: ty, $typeobject: expr, $checkfunction: path $(,$type_param: ident)*) => {
        pyobject_native_type!{$name, $typeobject, Some("builtin"), $checkfunction $(,$type_param)*}
    };
```
, and only the latter worked(macro invocation is rellay difficult in Rust... :confused:

Reply:
More like not very deterministic :disappointed: 
I made `$module` optional in both, but made a missing module be set as `None` instead of `Some("builtins")`, and left `Some("builtins")` in macro invocation of current native types. Omitting the module for type `x` would name the type `x` as well while calling the macro with `Some("builtins")` would name it `builtins.x`; I have no idea what this changes for the Python interpreter since I'm not that familiar with the CPython code.

Reply:
Hmm I think it's OK to make `Some("builtin")` default, since now this macro is used only by PyO3 and rust-numpy as far as I know.